python/lsst/ap/pipe/createApFakes.py

```diff
@@ -0,0 +1,300 @@
+            self.config.diskHLR: np.ones(nFakes, dtype="float"),
+            self.config.bulgeHLR: np.ones(nFakes, dtype="float"),
+            self.config.nDisk: np.ones(nFakes, dtype="float"),
+            self.config.nBulge: np.ones(nFakes, dtype="float"),
+            self.config.aDisk: np.ones(nFakes, dtype="float"),
+            self.config.aBulge: np.ones(nFakes, dtype="float"),
+            self.config.bDisk: np.ones(nFakes, dtype="float"),
+            self.config.bBulge: np.ones(nFakes, dtype="float"),
+            self.config.paDisk: np.ones(nFakes, dtype="float"),
+            self.config.paBulge: np.ones(nFakes, dtype="float")}
+
+        return Struct(
+            fakeCat=ParquetTable(dataFrame=pd.DataFrame(data=randData)))
+
+    def createRandomPositions(self, nFakes, boundingCircle):
+        """Create a set of spatially uniform randoms over the tract bounding
+        circle.
+
+        Parameters
+        ----------
+        nFakes : `int`
+            Number of fakes to create.
+        boundingCicle : `lsst.sphgeom.BoundingCircle`
+            Circle bound covering the tract.
+
+        Returns
+        -------
+        data : `dict`
+            Data dict containing:
+        """
+        # Create uniform random vectors on the sky around the north pole.
+        randVect = np.empty((nFakes, 3))
+        randVect[:, 2] = np.random.uniform(
+            np.cos(boundingCircle.getOpeningAngle().asRadians()),
+            1,
+            nFakes)
+        sinRawTheta = np.sin(np.arccos(randVect[:, 2]))
+        rawPhi = np.random.uniform(0, 2 * np.pi, nFakes)
+        randVect[:, 0] = sinRawTheta * np.cos(rawPhi)
+        randVect[:, 1] = sinRawTheta * np.sign(rawPhi)
+
+        # Compute the rotation matrix to move our random points to the
+        # correct location.
+        rotMatrix = self._createRotMatrix(boundingCircle)
+        randVect = np.dot(rotMatrix, randVect.transpose()).transpose()
+        decs = np.arcsin(randVect[:, 2])
+        ras = np.arctan2(randVect[:, 1], randVect[:, 0])
+
+        return {self.config.decColName: decs,
+                self.config.raColName: ras}
+
+    def _createRotMatrix(self, boundingCircle):
+        """Compute the 3d rotation matrix to rotate the dec=90 pole to the
+        center of the circle bound.
+
+        Parameters
+        ----------
+        boundingCircle : `lsst.sphgeom.BoundingCircle`
+             Circle bound covering the tract.
+
+        Returns
+        -------
+        rotMatrix : `numpy.ndarray`, (3, 3)
+            3x3 rotation matrix to rotate the dec=90 pole to the location of
+            the circle bound.
+        """
+        center = boundingCircle.getCenter()
+        cross = np.array([-center.y(),
+                          center.x(),
+                          0])
+        cross /= np.sqrt(cross[0] ** 2 + cross[1] ** 2 + cross[2] ** 2)
+        cosTheta = center.z()
+        sinTheta = np.sin(np.arccos(center.z()))
+
```

Comment:
I did not check that all the terms are correct in this matrix though I see it is tested with random angles (up to a rotation of ?); add a reference in a comment/docstring if this form of the rotation matrix is readily written down somewhere as a formula (I mean textbook, paper, WolframMathWorld, your notebook, tech note etc.).

Reply:
Wikipedia baby... https://en.wikipedia.org/wiki/Rotation_matrix the "Rotation from axis and angle" sub heading.